packages/ui/src/types/authMachine.ts

```diff
@@ -132,10 +143,17 @@ export type LoginMechanism = typeof LoginMechanismArray[number];
 
 export type SocialProvider = 'amazon' | 'apple' | 'facebook' | 'google';
 
-// other non-alias inputs that Cognito would require
-export type AuthInputNames = LoginMechanism | 'confirmation_code' | 'password';
-
-export type AuthInputAttributes = Record<AuthInputNames, InputAttributes>;
+// Auth fields that we provide default fields with
+export type AuthFieldsWithDefaults =
+  | LoginMechanism
+  | SignUpFieldsWithDefaults
+  | 'confirmation_code'
+  | 'password';
```

Comment:
This... can use a lot of clean ups. Punting it to post-GA per time restraints, but ideally we should use fewer variables to depict this. Also could implement `fieldHasDefaults(field: string): field is AuthFieldWithDefaults` to make sign up form field generation easier.

Reply:
👍